tests/test_remo.py

```diff
@@ -0,0 +1,260 @@
+        if empty:
+            mozilla_events_1 = read_file('data/remo_events_empty.json')
+
+        def request_callback(method, uri, headers):
+            offset = uri.split("offset=")[1].split("&")[0]
+            if offset == "0":
+                body = mozilla_events_1
+            else:
+                body = mozilla_events_2
+
+            HTTPServer.requests_http.append(httpretty.last_request())
+
+            return (200, headers, body)
+
+        httpretty.register_uri(httpretty.GET,
+                               MOZILLA_REPS_API_EVENTS,
+                               responses=[
+                                    httpretty.Response(body=request_callback)
+                               ])
+
+
+class TestReMoBackend(unittest.TestCase):
+    """ReMo backend tests"""
+
+    def test_initialization(self):
+        """Test whether attributes are initializated"""
+
+        remo = ReMo(MOZILLA_REPS_SERVER_URL, origin='test')
+
+        self.assertEqual(remo.url, MOZILLA_REPS_SERVER_URL)
+        self.assertEqual(remo.origin, 'test')
+        self.assertIsInstance(remo.client, ReMoClient)
+
+        # When origin is empty or None it will be set to
+        # the value in url
+        remo = ReMo(MOZILLA_REPS_SERVER_URL)
+        self.assertEqual(remo.url, MOZILLA_REPS_SERVER_URL)
+        self.assertEqual(remo.origin, MOZILLA_REPS_SERVER_URL)
+
+        remo = ReMo(MOZILLA_REPS_SERVER_URL, origin='')
+        self.assertEqual(remo.url, MOZILLA_REPS_SERVER_URL)
+        self.assertEqual(remo.origin, MOZILLA_REPS_SERVER_URL)
+
+    def __check_events_contents(self, events):
+        self.assertEqual(events[0]['data']['estimated_attendance'], 1000)
+        self.assertEqual(events[0]['data']['local_start'], '2012-06-06T10:00:00')
+        self.assertEqual(events[0]['data']['start'], '2012-06-06T02:00:00')
+        self.assertEqual(events[0]['origin'], MOZILLA_REPS_SERVER_URL)
+        self.assertEqual(events[0]['uuid'], 'e701d4ed3b954361383d678d2168a44307d7ff60')
+        self.assertEqual(events[0]['updated_on'], 1339326000.0)
+
+        if len(events) > 1:
+            self.assertEqual(events[1]['data']['estimated_attendance'], 50)
+            self.assertEqual(events[1]['data']['local_start'], '2012-06-09T09:00:00')
+            self.assertEqual(events[1]['data']['start'], '2012-06-09T07:00:00')
+            self.assertEqual(events[1]['origin'], MOZILLA_REPS_SERVER_URL)
+            self.assertEqual(events[1]['uuid'], 'e83e53b1f1039f62c66d2a977e23df3adf5bdc87')
+            self.assertEqual(events[1]['updated_on'], 1339344000.0)
+
+
+
+    @httpretty.activate
+    def test_fetch(self):
+        """Test whether the events are returned"""
+
+        HTTPServer.routes()
+
+        # Test fetch events with their reviews
+        remo = ReMo(MOZILLA_REPS_SERVER_URL)
+
+        events = [page for page in remo.fetch()]
+
+        self.assertEqual(len(events), 4)
+
+        self.__check_events_contents(events)
```

Comment:
Not sure about that. Checking the number of calls done? The params in each call?